types/react-image-magnify/index.d.ts

```diff
@@ -0,0 +1,228 @@
+// Type definitions for react-image-magnify 2.7
+// Project: https://github.com/ethanselzer/react-image-magnify
+// Definitions by: Sumit Parakh <https://github.com/sumitparakh>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import * as React from 'react';
+
+/**
+ * React Image Magnify
+ */
+declare const ReactImageMagnify: (props: ReactImageMagnifyProps & Readonly<{ children?: React.ReactNode }>) =>
```

Comment:
Instead of declaring this as a const with a function type, I suggest the following:

```ts
export default class ReactImageMagnify extends React.Component<ReactImageMagnifyProps> {}
```

Reply:
Problem is functional declaration is working in both stateless and statefull component.. But it stops working in stateless function component once i declare it like you suggested..  :/

Reply:
That's odd... can you add a test case demonstrating the issue that you're seeing? This will help make sure no one breaks it by making this change in the future.

Reply:
Hi @uniqueiniquity , I've added test cases for both stateless and stateful component.